APCAppCore/APCAppCore/Startup/APCAppDelegate.m

```diff
@@ -966,29 +960,26 @@ - (void) showAppropriateVC
 
 - (void)showPasscodeIfNecessary
 {
-    if (self.dataSubstrate.currentUser.isSignedIn && !self.isPasscodeShowing) {
+    if (self.dataSubstrate.currentUser.isSignedIn) {
         NSDate *lastUsedTime = [[NSUserDefaults standardUserDefaults] objectForKey:kLastUsedTimeKey];
-        
         if (lastUsedTime) {
-            NSTimeInterval timeDifference = [lastUsedTime timeIntervalSinceNow];
+            NSTimeInterval timeDifference = [[NSDate new] timeIntervalSinceDate:lastUsedTime];
             NSInteger numberOfMinutes = [self.dataSubstrate.parameters integerForKey:kNumberOfMinutesForPasscodeKey];
-            
-            if (timeDifference * -1 > numberOfMinutes * 60) {
-
-                [self showPasscode];
+            if (timeDifference > numberOfMinutes * 60) {
+                [self showPasscodeViewController];
             }
```

Comment:
I guess this approach won't defeat phone's system time change (rollback a certain period). Can we use `KERN_BOOTTIME ` instead? Even though kernel boot up time can reset, but it's more secure than system time.